src/librustc_mir/transform/const_prop.rs

```diff
@@ -501,19 +500,20 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         }
     }
 
-    fn report_panic_as_lint(&self, source_info: SourceInfo, panic: AssertKind<u64>) -> Option<()> {
-        // Somewhat convoluted way to re-use the CTFE error reporting code.
+    fn report_assert_as_lint(
+        &self,
+        lint: &'static lint::Lint,
+        source_info: SourceInfo,
+        message: &str,
```

Comment:
```suggestion
        message: &'static str,
```
The `struct_span_lint_hir` infra with `.build(message)` was intentionally setup in a way to discourage allocation of messages in case the lint doesn't fire. In all cases you have a `&'static str`, but `: &str` could invite doing `&format!(...)`.

Reply:
I don't feel like the cost of an allocation is relevant here but whatever.